runtime/autoload/freebasic.vim

```diff
@@ -0,0 +1,41 @@
+" Vim filetype plugin file
+" Language:	FreeBASIC
+" Maintainer:	Doug Kearns <dougkearns@gmail.com>
+" Last Change:	2021 Mar 16
+
+" Dialects can be one of fb, qb, fblite, or deprecated
+" Precedence is forcelang > #lang > lang
+function! freebasic#GetDialect() abort
+  if exists("g:freebasic_forcelang")
+    return g:freebasic_forcelang
+  endif
+
+  if exists("g:freebasic_lang")
+    let dialect = g:freebasic_lang
+  else
+    let dialect = "fb"
+  endif
+
+  " override with #lang directive or metacommand
+
+  let skip = "has('syntax_items') && synIDattr(synID(line('.'), col('.'), 1), 'name') =~ 'Comment$'"
+  let pat = '\c^\s*\%(#\s*lang\s\+\|''\s*$lang\s*:\s*\)"\([^"]*\)"'
+
+  let save_cursor = getcurpos()
+  call cursor(1, 1)
+  let lnum = search(pat, 'n', '', '', skip)
```

Comment:
Hmm, actually this needs https://github.com/vim/vim/commit/adc17a5f9d207fd1623fd923457a46efc9214777

Reply:
Ouch :/

Reply:
Very sneaky too! I'll see if it can be ported

Reply:
Or the `skip` be avoided...

Reply:
Maybe. At first glance seems to be used to ignore language directives that have been commented out.

Reply:
We could just ignore the skip for now.
Same is done if `syntax_items` aren't supported, so it's probably not the end of the world :P

Reply:
Makes sense; I left the line commented out with a mention of the missing patch so it can be revisited.